### PR TITLE
Count store from kernel API

### PR DESCRIPTION
**Pull request by pontusmelke**
Move (or rather copy, but will be removed soon) count store operations to kernel api which allows us to get rid of all dependency on `ReadOperation` from the compiled runtime

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java [hunk 1]
```diff
@@ -115,6 +121,37 @@ else if ( txState.nodeIsAddedInThisTx( reference ) )
...
+        {
+            CountsRecordState counts = new CountsRecordState();
+            try
+            {
+                TransactionState txState = ktx.txState();
+                txState.accept( new TransactionCountingStateVisitor( EMPTY, storeReadLayer,
+                        statement, txState, counts ) );
+                if ( counts.hasChanges() )
+                {
+                    count += counts.nodeCount( labelId, newDoubleLongRegister() ).readSecond();
```
Comment (SaschaPeukert) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java:138]: Except for that ```readSecond()``` part, everything else is really good to read. Maybe renaming that method could help to better express what it does?

File: community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/TransactionBoundGraphStatistics.scala [hunk 2]
```diff
@@ -21,17 +21,19 @@ package org.neo4j.cypher.internal.runtime.interpreted
...
+import org.neo4j.internal.kernel.api.Read
 import org.neo4j.kernel.api.ReadOperations
 import org.neo4j.kernel.api.exceptions.index.IndexNotFoundKernelException
 
 object TransactionBoundGraphStatistics {
-  def apply(ops: ReadOperations) = new StatisticsCompletingGraphStatistics(new BaseTransactionBoundGraphStatistics(ops))
+  def apply(read: Read, ops: ReadOperations) = new StatisticsCompletingGraphStatistics(new BaseTransactionBoundGraphStatistics(read, ops))
 
-  private class BaseTransactionBoundGraphStatistics(operations: ReadOperations) extends GraphStatistics with IndexDescriptorCompatibility {
+  //TODD remove ReadOperations when schema ops are ported
```
Comment (SaschaPeukert) [inline community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/TransactionBoundGraphStatistics.scala]: Who is Todd? :-D

**Review by SaschaPeukert — CHANGES_REQUESTED** [review]
Looks good! I only have two minor nit-picks

File: community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java (hunk 1 above)
Comment (pontusmelke) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java:138]: This is the API of the underlying count store, don't want to change that API as part of this PR.

File: community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/TransactionBoundGraphStatistics.scala (hunk 2 above)
Comment (pontusmelke) [reply community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/TransactionBoundGraphStatistics.scala]: oh yes TODD, I totally meant to write that :)

**Review by SaschaPeukert — APPROVED** [review]

**Pull request merged**